Mariana/layers.py

```diff
@@ -540,29 +536,29 @@ def setCustomTheanoFunctions(self) :
 
 		self.classificationAccuracy = MWRAP.TheanoFunction("classificationAccuracy", self, [("accuracy", clasAcc)], { "targets" : self.targets }, allow_input_downcast=True)
 		self.predictionAccuracy = MWRAP.TheanoFunction("predictionAccuracy", self, [("accuracy", predAcc)], { "targets" : self.targets }, allow_input_downcast=True)
-		
+
 		self.trainAndAccuracy = MWRAP.TheanoFunction("trainAndAccuracy", self, [("score", self.cost), ("accuracy", clasAcc)], { "targets" : self.targets },  updates = self.updates, allow_input_downcast=True)
 		self.testAndAccuracy = MWRAP.TheanoFunction("testAndAccuracy", self, [("score", self.testCost), ("accuracy", predAcc)], { "targets" : self.targets }, allow_input_downcast=True)
-		
-class Regression(Output_ABC) :
+
+class Regression(WeightBiasOutput_ABC) :
 	"""For regressions, works great with a mean squared error cost"""
 	def __init__(self, nbOutputs, activation, learningScenario, costObject, name = None, **kwargs) :
-		Output_ABC.__init__(self, nbOutputs, activation = activation, learningScenario = learningScenario, costObject = costObject, name = name, **kwargs)
+		super(Regression, self).__init__(nbOutputs, activation=activation, learningScenario=learningScenario, costObject=costObject, name=name, **kwargs)
 		self.targets = tt.matrix(name = "targets")
 
-class Autoencode(Output_ABC) :
+class Autoencode(WeightBiasOutput_ABC) :
 	"""An auto encoding layer. This one takes another layer as inputs and tries to reconstruct its activations.
-	You could achieve the same result with a Regresison layer, but this one has the advantage of not needing to be fed specific inputs"""
+	You could achieve the same result with a Regression layer, but this one has the advantage of not needing to be fed specific inputs"""
 
-	def __init__(self, targetLayerName, activation, learningScenario, costObject, name = None, **kwargs) :
-		Output_ABC.__init__(self, None, activation = activation, learningScenario = learningScenario, costObject = costObject, name = name, **kwargs)
+	def __init__(self, targetLayerName, activation, learningScenario, costObject, name=None, **kwargs) :
+		super(Autoencode, self).__init__(None, activation=activation, learningScenario=learningScenario, costObject=costObject, name=name, **kwargs)
 		self.targetLayerName = targetLayerName
 
 	def _whateverFirstInit(self) :
 		self.nbOutputs = self.network[self.targetLayerName].nbOutputs
 		self.targets = self.network[self.targetLayerName].outputs
-		
+
 	def setCustomTheanoFunctions(self) :
-		Output_ABC.setCustomTheanoFunctions(self)
+		super(Autoencode, self).setCustomTheanoFunctions()
```

Comment:
My guess is that it is not. But In the worst case scenario it is just a call to a function that does nothing. I will investigate.